### ISSUE TITLE
Feature Request: Support setting notification "interruptionLevel"

### ISSUE THREAD

**Issue by npars** [ptsochantaris/trailer#478]
I'm in a fairly large Github org and sometimes miss `@mention` notifications that are directed at me. It would be nice if trailer supported setting a different [`interruptionLevel`](https://developer.apple.com/documentation/usernotifications/unnotificationinterruptionlevel) for notifications based on some criteria (contains an `@mention`, your PR, etc.).

I poked around the codebase and noticed that Trailer is using `NSUserNotification` which is now marked deprecated and does not support setting the `interruptionLevel` like the newer [`UNNotification`](https://developer.apple.com/documentation/usernotifications) API. So unfortunately I think this would be a non-trivial change.

**Comment by ptsochantaris** [comment]
Hi @npars and thank you for this suggestion - it totally makes sense. You are totally correct about the UNNotification framework issue, that's because Trailer on the Mac before UNNotification showed up about 3 iOS versions later :D 

However this is a good opportunity to modernise that aspect of it I think. I've mostly demurred from it because it involves migrating existing notifications to action categories instead of the "old" method of simply attaching actions to notifications, but the more I think about things like this and things like App Intent support for automation and so forth, the more I think it's better to start moving towards cleaning that stuff up.

TL;DR (whoops, too late) though, I think this makes total sense as a feature, and I'll put it on the TODO list.

**Comment by ptsochantaris** [comment]
Hi @npars - So, it's taken a few weeks of cleaning up various things, and part of that was moving to UNNotification and some good cleanup to integrate the Mac and iOS code paths there. Felt good 😄 

And now we can build this feature. Since the "critical" level of the notification isn't accessible by mere mortals and meant for specific apps like emergency or government notifications, I assume what we want is to be able to chose between the thee levels:
- High (aka "time sensitive")
- Default (current - aka "active")
- Low (aka "passive")

The question now is more about criteria, I think. I can imagine a few workflows where I'd love for instances to mark notifications for a whole repo as "passive" and others as "active".

#### Repo
I'd definitely want a criterion to set notification levels based on the related repo, for example I'd want to promote notifications related to issues in Trailer no matter what, and demote all issues created in the Swift language repo (as I'm not an active participant there).

#### Section
I imagine I'd want to promote and demote a notification based on the section of the related item is. e.g. In my case I'd want to always promote notifications for items in "mine" and always demote notifications about items in "all".

#### Notification subject
I'd also want notifications to be "promoted" or "demoted" based on their type. For instance in my case I'd "promote" ones related to review approvals and demote ones about CI statuses.

#### Free-text?
Does it make sense to promote or demote a notification based on some text? I'm inclined to say yes - I'm thinking in some particular context I may want to immediately know about any notification about some arbitrary subject or project mentioned in the item related to the notification. This criterion would search the same data that the menu filters would search (e.g. title, body, labels, etc)

Those are the things that immediately spring to mind, but do let me know your thoughts as well. Can't promise I'll implement everything, but definitely worth starting off from a position of more choices than fewer!

**Comment by npars** [comment]
The big one for me would be Free-text / Keyword based (much like Slack). That way I could add my user `@npars` and have those notifications "promoted". 

After that I think threaded replies would be very useful. If someone replies to a comment I've made (or replies in a thread I've participated in) I want to respond to those as fast as possible.
